### PR TITLE
Taint script verify

### PR DESCRIPTION
**Pull request by jonathanKingston**

### CODE REVIEW

File: src/features/runtime-checks/script-overload.js [hunk 1]
```diff
@@ -132,7 +201,7 @@ export function wrapScriptCodeOverload (code, config) {
         processedConfig[key] = processAttr(value)
     }
     // Don't do anything if the config is empty
-    if (Object.keys(processedConfig).length === 0) return code
+    // if (Object.keys(processedConfig).length === 0) return code
```
Comment (jonathanKingston) [inline src/features/runtime-checks/script-overload.js:204]: Fix me.

File: src/features/runtime-checks.js [hunk 2]
```diff
@@ -471,6 +470,21 @@ function overrideCreateElement () {
...
+function getTaintFromScope (scope, args) {
+    try {
+        scope = args.callee.caller
+    } catch {
+        console.log('taint: failed to get callers scope', args, scope)
+    }
+    return hasTaintedMethod(scope)
+}
+
+function injectGenericOverloads () {
```
Comment (dharb) [inline src/features/runtime-checks.js:482]: Assuming you intend to follow up on this in a subsequent PR?

File: src/features/runtime-checks/script-overload.js [hunk 3]
```diff
@@ -1,4 +1,66 @@
-import { processAttr } from '../../utils.js'
+import { processAttr, getContextId } from '../../utils.js'
+
+const globalStates = new Set()
+
+function generateUniqueID () {
+    const debug = true
+    if (debug) {
```
Comment (dharb) [inline src/features/runtime-checks/script-overload.js:7]: What's your plan here?

File: src/features/runtime-checks/script-overload.js [hunk 4]
```diff
@@ -1,4 +1,66 @@
...
+
+function addTaint () {
+    const contextID = generateUniqueID()
+    if ('duckduckgo' in navigator &&
+        navigator.duckduckgo &&
+        typeof navigator.duckduckgo === 'object' &&
+        'taints' in navigator.duckduckgo &&
+        navigator.duckduckgo.taints instanceof Set) {
+        if (document.currentScript) {
+            // @ts-expect-error - contextID is undefined on cuttentScript
```
Comment (dharb) [inline src/features/runtime-checks/script-overload.js:22]: nit:
```suggestion
            // @ts-expect-error - contextID is undefined on currentScript
```

File: src/features/runtime-checks/script-overload.js [hunk 5]
```diff
@@ -1,4 +1,66 @@
-import { processAttr } from '../../utils.js'
+import { processAttr, getContextId } from '../../utils.js'
+
+const globalStates = new Set()
+
+function generateUniqueID () {
+    const debug = true
```
Comment (dharb) [inline src/features/runtime-checks/script-overload.js:6]: Do you intend to leave this?

**Review by dharb — APPROVED** [review]
All looks pretty good from where I'm standing, left a few minor comments. Is your intent is to leave the handful of `console.log` statements?

File: src/features/runtime-checks.js (hunk 2 above)
Comment (jonathanKingston) [reply src/features/runtime-checks.js:482]: Yeah testing that on real sites now to see the impacts.

File: src/features/runtime-checks/script-overload.js (hunk 5 above)
Comment (jonathanKingston) [reply src/features/runtime-checks/script-overload.js:6]: Not one really, neither mode is bad tbh. I'll set it to false, getting the debugger state here is a bit of a pain that can wait for another PR.

**Comment by jonathanKingston** [comment]
> Is your intent is to leave the handful of console.log statements?

Nah sorry was working on the follow up patch where they were removed. I've dropped them from here too. Thanks.

**Pull request merged**